src/main/java/net/minecraftforge/fluids/FluidStack.java

```diff
@@ -84,17 +86,46 @@ public static FluidStack loadFluidStackFromNBT(NBTTagCompound nbt)
         {
             return null;
         }
-        if (!nbt.hasKey("FluidName", Constants.NBT.TAG_STRING))
+
+        FluidStack stack;
+        if (nbt.hasKey("Fluid", Constants.NBT.TAG_STRING))
         {
-            return null;
+            ResourceLocation fluid = new ResourceLocation(nbt.getString("Fluid"));
+            if (!ForgeRegistries.FLUIDS.containsKey(fluid))
+            {
+                return null;
+            }
+            else
+            {
+                stack = new FluidStack(ForgeRegistries.FLUIDS.getValue(fluid), nbt.getInteger("Amount"));
+            }
         }
-
-        String fluidName = nbt.getString("FluidName");
-        if (FluidRegistry.getFluid(fluidName) == null)
+        //TODO Drop legacy support for 'FluidNane' in 1.14 or 1.15, as it will be outdated 
+        else if (nbt.hasKey("FluidName", Constants.NBT.TAG_STRING))
+        {
+            String fluidName = nbt.getString("FluidName").toLowerCase();
+            Fluid fluid = null;
+            for (Fluid f : ForgeRegistries.FLUIDS.getValues())
+            {
+                if (f.getRegistryName().getResourcePath().equals(fluidName));
+                {
```

Comment:
Would it be alright to add this before it checks the registry?
`if (FluidDictionary.nameExists(fluidName))
       return FluidDictionary.getFirstFluid(fluidName));`